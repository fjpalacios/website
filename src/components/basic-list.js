import * as React from 'react'
import './basic-list.scss'

const BasicList = ({ data }) => {
  return (
    <>
      {data.map((item, key) => {
        const Name = () => {
          if (item.url) {
            return <a href={item.url}>{item.name}</a>
          }

          return <span>{item.name}</span>
        }

        const RoleAndLocation = () => {
          if (item?.role) {
            return (
              <>
                <div className="basic-list__role">{item.role}</div>
                <div className="basic-list__location--with-role">{item.location}</div>
              </>
            )
          } else {
            return <div className="basic-list__location">{item.location}</div>
          }
        }

        return (
          <div key={key} className="basic-list">
            <div className="basic-list__name">
              <Name />
            </div>
            <div className="basic-list__dates">{item.dates}</div>
            <RoleAndLocation />
            <div className="basic-list__desc" dangerouslySetInnerHTML={{ __html: item.desc }}></div>
            <div className="basic-list__extendedDesc" dangerouslySetInnerHTML={{ __html: item.extendedDesc }}></div>
          </div>
        )
      })}
    </>
  )
}

export default BasicList
