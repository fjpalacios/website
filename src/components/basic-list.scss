.basic-list {
  display: grid;
  grid-template-areas:
    'name dates'
    'role location-role'
    'location .'
    'desc desc'
    'extendedDesc extendedDesc';
  margin-bottom: 20px;
  padding: 0 20px;

  @include print {
    margin-bottom: 10px;
    padding: 0 10px;
  }

  &__name {
    grid-area: name;
    text-transform: uppercase;
  }

  &__role {
    grid-area: role;
    font-weight: bold;
    text-transform: uppercase;
  }

  &__location {
    grid-area: location;

    &--with-role {
      display: grid;
      grid-area: location-role;
      justify-content: end;
    }
  }

  &__dates {
    display: grid;
    grid-area: dates;
    justify-content: end;
  }

  &__desc {
    grid-area: desc;

    @include screen {
      display: none;
    }
  }

  &__extendedDesc {
    grid-area: extendedDesc;

    @include print {
      display: none;
    }
  }
}