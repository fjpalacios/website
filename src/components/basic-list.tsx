import './basic-list.scss'
import React, { FunctionComponent, ReactElement } from 'react'

export type basicList = {
  name: string
  url?: string
  location: string
  dates: string
  desc: string
}

type basicListProps = {
  data: basicList[]
}

export const BasicList: FunctionComponent<basicListProps> = ({
  data,
}): ReactElement => {
  return (
    <>
      {data.map((item, key) => {
        const Name = (): ReactElement => {
          if (item.url) {
            return <a href={item.url}>{item.name}</a>
          }

          return <span>{item.name}</span>
        }

        return (
          <div key={key} className="basic-list">
            <div className="basic-list__name">
              <Name />
            </div>
            <div className="basic-list__dates">{item.dates}</div>
            <div className="basic-list__location">{item.location}</div>
            <div
              className="basic-list__desc"
              dangerouslySetInnerHTML={{ __html: item.desc }}
            ></div>
          </div>
        )
      })}
    </>
  )
}
