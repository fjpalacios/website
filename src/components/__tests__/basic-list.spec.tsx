import { basicList, BasicList } from '../basic-list'
import createComponentWithIntl from '../../../__helpers__/i18n-component'
import React from 'react'

describe('BasicList', () => {
  const data: basicList[] = [
    {
      name: 'Lorem ipsum',
      location: 'Location',
      dates: '10/07/1856',
      desc: 'Lorem ipsum dolor sit amet, consectetur adipiscing elit.',
    },
  ]

  it('should render properly', () => {
    const basicList = createComponentWithIntl(<BasicList data={data} />)
    expect(basicList).toMatchSnapshot()
  })

  it('should show the name', () => {
    const basicList = createComponentWithIntl(<BasicList data={data} />)
    const name = basicList.find('.basic-list__name span').text()
    expect(name).toBe(data[0].name)
  })

  it('should show a url if it exists', () => {
    data[0].url = 'https://duckduckgo.com'
    const basicList = createComponentWithIntl(<BasicList data={data} />)
    const name = basicList.find('.basic-list__name a').text()
    expect(name).toBe(data[0].name)
    data[0].url = undefined
  })

  it('should show the location', () => {
    const basicList = createComponentWithIntl(<BasicList data={data} />)
    const location = basicList.find('.basic-list__location').text()
    expect(location).toBe(data[0].location)
  })

  it('should show the dates', () => {
    const basicList = createComponentWithIntl(<BasicList data={data} />)
    const dates = basicList.find('.basic-list__dates').text()
    expect(dates).toBe(data[0].dates)
  })

  it('should show the description', () => {
    const basicList = createComponentWithIntl(<BasicList data={data} />)
    const desc = basicList.find('.basic-list__desc').text()
    expect(desc).toBe(data[0].desc)
  })
})
