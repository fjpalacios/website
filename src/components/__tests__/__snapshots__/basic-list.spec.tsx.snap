// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BasicList should render properly 1`] = `
<IntlProvider
  defaultFormats={Object {}}
  defaultLocale="en"
  formats={Object {}}
  locale="en"
  messages={
    Object {
      "about": Object {
        "education": "Education",
        "experience": "Experience",
        "freelance": "Freelance work",
        "me": "About me",
        "projects": "Side projects",
        "talks": "Talks / workshops",
        "volunteering": "Volunteering",
      },
      "menu": Object {
        "switchTheme": "Switch Theme",
        "toEnglish": "Switch to English language",
        "toSpanish": "Switch to Spanish language",
      },
      "subtitle": "Software Developer",
      "title": "Francisco Javier Palacios PÃ©rez",
    }
  }
  onError={[Function]}
  textComponent={Symbol(react.fragment)}
>
  <BasicList
    data={
      Array [
        Object {
          "dates": "10/07/1856",
          "desc": "Lorem ipsum dolor sit amet, consectetur adipiscing elit.",
          "location": "Location",
          "name": "Lorem ipsum",
        },
      ]
    }
  >
    <div
      className="basic-list"
      key="0"
    >
      <div
        className="basic-list__name"
      >
        <Name>
          <span>
            Lorem ipsum
          </span>
        </Name>
      </div>
      <div
        className="basic-list__dates"
      >
        10/07/1856
      </div>
      <div
        className="basic-list__location"
      >
        Location
      </div>
      <div
        className="basic-list__desc"
        dangerouslySetInnerHTML={
          Object {
            "__html": "Lorem ipsum dolor sit amet, consectetur adipiscing elit.",
          }
        }
      />
    </div>
  </BasicList>
</IntlProvider>
`;
